components/search/AutoSuggest.vue

```diff
@@ -0,0 +1,233 @@
+        };
+      }
+    },
+
+    watch: {
+      options() {
+        this.isActive = !!this.options && this.query.length > 2;
+      },
+      '$route.query'() {
+        this.onSearchablePage ? this.query = this.$store.state.search.query : this.query = '';
+      }
+    },
+
+    mounted() {
+      document.addEventListener('keyup', this.navigateDropdown);
+      document.addEventListener('mouseup', this.clickOutside);
+      this.query = this.$store.state.search.query;
+    },
+
+    methods: {
+      clickOutside(event) {
+        const isChild = this.$el.contains(event.target);
+
+        if (!isChild) {
+          this.isActive = false;
+        }
+      },
+      updateInputWithHighlighted() {
+        this.$nextTick(() => {
+          const highlighted = document.querySelector('.highlighted');
+          if (!highlighted) return;
+          this.query = highlighted.getAttribute('value');
+        });
+      },
+      navigateDropdown() {
+        switch (event.keyCode) {
+        case 38:
+          if (this.focus === null) {
+            this.focus = 0;
+          } else if (this.focus > 0) {
+            this.focus--;
+          } else if (this.focus === 0) {
+            this.focus = null;
+          }
+          this.updateInputWithHighlighted();
+          break;
+        case 40:
+          if (this.focus === null) {
+            this.focus = 0;
+          } else if (this.focus < Object.keys(this.options).length - 1) {
+            this.focus++;
+          }
+          this.updateInputWithHighlighted();
+          break;
+        }
+      },
+
+      activateDropdown() {
+        return !this.isActive && this.options;
+      },
+
+      async submitForm() {
+        const newRouteQuery = { ...this.$route.query, ...{ query: this.query, page: 1, view: this.view } };
+        const newRoute = { path: this.routePath, query: newRouteQuery };
+        this.isActive = false;
+        await this.$router.push(newRoute);
+      },
+
+      // FAKE DATA
+      async getSuggestions() {
+        setTimeout(() => {
+          this.options = {
+            'http://data.europeana.eu/concept/base/83': 'Hello',
+            'http://data.europeana.eu/concept/base/94': 'By Hello'
+          };
+        }, 500);
+      }
+    }
+  };
+</script>
+
+<style lang="scss" scoped>
+  @import "./assets/scss/variables.scss";
+
+  .auto-suggest {
+    position: relative;
+
+    &-dropdown {
+      position: absolute;
+      top: 50px;
+      width: 100%;
+      z-index: 20;
+
+      a.list-group-item {
+        border: 0;
+        box-shadow: none;
+        padding: .75rem 1.25rem;
+
+        &.highlighted {
+          background-color: #f1f1ee;
+        }
+      }
+    }
+  }
+
+  .input-group {
+    width: 100%;
+
+    .input-group-prepend {
+      align-items: center;
+      background-color: $lightgrey;
+      padding-left: .75rem;
+      padding-right: .1rem;
+      border-radius: 0.375rem 0 0 0.375rem;
+    }
```

Comment:
Is this still used?

Reply:
It does now as its required for the entity page. I have just stopped the autosuggest functionality when on the entity page